var/spack/repos/builtin/packages/cantera/package.py

```diff
@@ -0,0 +1,197 @@
+        ]
+
+        # BLAS/LAPACK support
+        if '+lapack' in spec:
+            options.extend([
+                'blas_lapack_libs=lapack,blas',
+                'blas_lapack_dir={0}'.format(spec['lapack'].prefix.lib)
+            ])
+
+        # Threadsafe build, requires Boost
+        if '+threadsafe' in spec:
+            options.extend([
+                'build_thread_safe=yes',
+                'boost_inc_dir={0}'.format(spec['boost'].prefix.include),
+                'boost_lib_dir={0}'.format(spec['boost'].prefix.lib),
+                'boost_thread_lib=boost_thread-mt,boost_system-mt'
+            ])
+        else:
+            options.append('build_thread_safe=no')
+
+        # Sundials support
+        if '+sundials' in spec:
+            options.extend([
+                'use_sundials=y',
+                'sundials_include={0}'.format(spec['sundials'].prefix.include),
+                'sundials_libdir={0}'.format(spec['sundials'].prefix.lib),
+                'sundials_license={0}'.format(
+                    join_path(spec['sundials'].prefix, 'LICENSE'))
+            ])
+        else:
+            options.append('use_sundials=n')
+
+        # Python module
+        if '+python' in spec:
+            options.extend([
+                'python_package=full',
+                'python_cmd={0}'.format(
+                    join_path(spec['python'].prefix.bin, 'python')),
+                'python_array_home={0}'.format(spec['py-numpy'].prefix)
+            ])
+            if spec['python'].satisfies('@3'):
+                options.extend([
+                    'python3_package=y',
+                    'python3_cmd={0}'.format(
+                        join_path(spec['python'].prefix.bin, 'python')),
+                    'python3_array_home={0}'.format(spec['py-numpy'].prefix)
+                ])
+            else:
+                options.append('python3_package=n')
+        else:
+            options.append('python_package=none')
+            options.append('python3_package=n')
+
+        # Matlab toolbox
+        if '+matlab' in spec:
+            options.extend([
+                'matlab_toolbox=y',
+                'matlab_path={0}'.format(spec['matlab'].prefix)
+            ])
+        else:
+            options.append('matlab_toolbox=n')
+
+        scons('build', *options)
+
+        if '+python' in spec:
+            # Tests will always fail if Python dependencies aren't built
+            scons('test', parallel=False)
+
+        scons('install')
+
+        self.filter_compilers()
+
+    def filter_compilers(self):
+        """Run after install to tell the Makefile and SConstruct files to use
+        the compilers that Spack built the package with.
+
+        If this isn't done, they'll have CC, CXX, F77, and FC set to Spack's
+        generic cc, c++, f77, and f90. We want them to be bound to whatever
+        compiler they were built with."""
+
+        kwargs = {'ignore_absent': True, 'backup': False, 'string': True}
```

Comment:
So this is weird. Setting `'backup': False` works for all of the CC files, but not for CXX, F77, or FC. Any idea what's wrong here? Is kwargs somehow getting reset?